plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs

```diff
@@ -164,24 +164,25 @@ rules recorder plId = do
               logWith recorder Debug (LogDebugStanEnvVars env)
               seTomlFiles <- liftIO $ usedTomlFiles useDefConfig (stanArgsConfigFile stanArgs)
 
+              -- Note that Stan works in terms of relative paths, but the HIE come in as absolute. Without
+              -- making its path relative, the file name(s) won't line up with the associated Map keys.
+              relativeHsFilePath <- liftIO $ makeRelativeToCurrentDirectory $ fromNormalizedFilePath file
+              let hieRelative = hie{hie_hs_file=relativeHsFilePath}
+
               (cabalExtensionsMap, checksMap, confIgnored) <- case configTrial of
                   FiascoL es -> do
                       logWith recorder Development.IDE.Warning (LogWarnConf es)
                       pure (Map.empty,
                             HM.fromList [(LSP.fromNormalizedFilePath file, inspectionsIds)],
                             [])
```

Comment:
Oh, I almost forgot to ask: I wasn't sure about desired behavior if we hit this branch.

Of course I could change this file path to `relativeHsFilePath` and even build a `cabalExtensionsMap`, I think... but when I checked the corresponding flow in [Stan-proper](https://github.com/kowainik/stan/blob/main/src/Stan.hs#L90), they seem to bail entirely in the case of a `Fiasco` here. (Assuming I'm reading the code correctly.)

Reply:
Yes, you read the code correctly. Stan stops completely with an error. 

We could use this opportunity to define what the plugin should do. When I added the capability for reading the config, I left the previous behavior (stan runs without config) on the branch where something is wrong here, however we could instead stop the analysis completely.

The latter could be preferred in the hypothetical case where there is a bad config, and the user would just prefer to ignore possibly annoying suggestions (that would be ignored taking into account a well formed .stan.toml).

However, I'm not partial to any choice. Any suggestions @michaelpj?

Reply:
Ah, I see. Yeah, I could see a good argument for going either way.

As you say, it could be annoying for a user to see a bunch of irrelevant diagnostics. But on the other hand, it could be a bad experience for the plugin to just fail silently and for the user to see nothing at all. They could break a config and they might not even notice. (Talking through it, that latter case might be worse in the end?)

In any case, if the intended previous behavior was for it to still run sans config, I'd be fine just defaulting to that. (In which case, I'd have to make some minor tweaks to the code, fix the mappings there as well.)

Reply:
Let's stick to running diagnostics anyway and building the `cabalExtensionsMap`, so that it gets taken into account!

Reply:
> Let's stick to running diagnostics anyway and building the cabalExtensionsMap, so that it gets taken into account!

Done! Pushed up https://github.com/haskell/haskell-language-server/pull/4023/commits/e6a85206dccc08fec9cfe907e308987b52c3b8d5 to take care of that.

Interestingly, I couldn't "organically" hit that code path from the plugin no matter what I tried -- I wound up just forcing the fiasco for testing purposes.

Seems like Stan will always either 1) throw an IO exception (in the case of a TOML parse error), or 2) recover and return some variety of partial/default config. But that path works properly now, just in case that should ever change :smile: 

On the plus side (?!), the "invalid config" case is extremely loud and obvious, thanks to its exception-throwing. Basically brings everything screeching to a halt. Here's how it looks for me in neovim:

![hls-stan-toml-parse-error](https://github.com/haskell/haskell-language-server/assets/132365/78f2b976-18f4-42c5-9765-34e60e17801f)